src/Sidenav/context.ts

```diff
@@ -39,17 +39,24 @@ export const propsDeepSearch = (
       }
     } else {
       for (const prop in theObject) {
-        if (theObject.hasOwnProperty(prop)) {
-          if (prop === targetProp) {
-            if (theObject[prop] === targetValue) {
-              finalResults.push(theObject);
+        if (
```

Comment:
Could we do something like ```
const acceptablePropTypes = ['$$typeof', 'children', ...etc];
if (acceptablePropTypes.includes(prop)) {}
```

Reply:
Thanks for pointing that out. I was so happy to find the work-around that I didn't even think of refactoring.